packages/frontend-2/lib/workspaces/composables/management.ts

```diff
@@ -522,3 +598,71 @@ export const useSetDefaultWorkspaceRegion = () => {
     return res?.data?.workspaceMutations.setDefaultRegion
   }
 }
+
+export function useSsoAuth(workspaceSlug: string) {
+  const logger = useLogger()
+  const ssoError = ref<string | null>(null)
+
+  const { hasSsoEnabled, needsSsoLogin, error } = useWorkspaceSso({
+    workspaceSlug
+  })
+
+  if (error.value) {
+    logger.error('SSO check failed:', error.value)
+    ssoError.value = 'Failed to check workspace SSO requirements'
+  } else if (hasSsoEnabled.value && needsSsoLogin.value) {
+    ssoError.value = 'You need to sign in with SSO to access this workspace.'
+  }
+
+  return { ssoError }
+}
+
+export function useWorkspaceSsoDelete() {
+  const apiOrigin = useApiOrigin()
+  const { triggerNotification } = useGlobalToast()
+  const mixpanel = useMixpanel()
+
+  const deleteSsoProvider = async (workspaceSlug: string) => {
+    try {
+      const res = await fetch(
+        new URL(`/api/v1/workspaces/${workspaceSlug}/sso`, apiOrigin),
```

Comment:
@cdriesler Why are a lot of these REST endpoints? Feels like a few of these could just be GQL queries/mutations

Reply:
I think this is (should be) the only one left that needs to be converted to gql. Predates your flag, but will happen. 👍